airbyte-integrations/connectors/source-surveycto/source_surveycto/source.py

```diff
@@ -109,9 +109,26 @@ def read_records(self, *args, **kwargs) -> Iterable[Mapping[str, Any]]:
 
 # Source
 class SourceSurveycto(AbstractSource):
+    
+    def _base64_encode(self, string: str) -> str:
+        return base64.b64encode(string.encode("ascii")).decode("ascii")
+    
     def check_connection(self, logger, config) -> Tuple[bool, any]:
-        return True, None
 
+        server_name=config['server_name']
+        form_id=config['form_id'][0]
+        url=f"https://{server_name}.surveycto.com/" + f"api/v2/forms/data/wide/json/{form_id}?date=Jan 09, 2100 00:00:00 AM"
+        user_name_password = f"{config['username']}:{config['password']}"
+        auth_token = self._base64_encode(user_name_password)
+        headers = {"Authorization": "Basic " + auth_token}
+        response = requests.get(url, headers=headers)
```

Comment:
I don't think this is a good implementation for the check connection.
@avirajsingh7 please check https://discuss.airbyte.io/t/how-the-airbyte-cdk-check-command-works/4589 to see how other connectors usually do this step.